packages/runtime/src/routeGeneration/templateHelpers.ts

```diff
@@ -154,8 +164,8 @@ export class ValidationService {
     return value;
   }
 
-  public validateInt(name: string, value: any, fieldErrors: FieldErrors, validators?: IntegerValidator, parent = '') {
-    if (!validator.isInt(String(value))) {
+  public validateInt(name: string, value: any, fieldErrors: FieldErrors, isBodyParam: boolean, swaggerConfig: AdditionalProps, validators?: IntegerValidator, parent = '') {
+    if ((isBodyParam && swaggerConfig.bodyCoercion === false && typeof value !== 'number') || !validator.isInt(String(value))) {
```

Comment:
Can you extract this to a fn?

Reply:
At least the first 2 conditions into fn shouldCoerce, possibly with a callback for the third condition(s)

Reply:
I've extracted a function, what do you think?

Reply:
I've extracted this to a function, what do you think?